python/pyspark/sql/session.py

```diff
@@ -611,8 +612,8 @@ def _inferSchema(
         :class:`pyspark.sql.types.StructType`
         """
         first = rdd.first()
-        if not first:
-            raise ValueError("The first row in RDD is empty, " "can not infer schema")
+        if isinstance(first, Sized) and len(first) == 0:
```

Comment:
Or maybe shall we just remove this check away? We take 100 rows and infer so I don't think take 1 + failfast is super beneficial but it burdens more overhead.

Reply:
what about filtering out the `None` rows:

```
first  = rdd.filter( no None cond ).first()
```

Reply:
we can do .. but I started to think that this check isn't very useful ... especially given that it will trigger another Spark job that can potentially be very expensive.

Reply:
+1, the `first` job seems unnecessary:

1, `if samplingRatio is None`, we should directly take first 100 non-None rows, check on the 100 rows and infer the schema;
2, `if samplingRatio is not None`, the `_merge_type` job should be enough. If we need to check the count, we can add a `count` into `_merge_type`.